### PR TITLE
Testunit filter cases

### PR DESCRIPTION
**Pull request by mazenmmbadr**

### CODE REVIEW

File: ui_testing/testcases/basic_tests/test04_testunits.py [hunk 1]
```diff
@@ -1719,4 +1705,168 @@ def test041_hide_all_table_configurations(self):
...
+        data_to_filter_with = self.test_unit_api.get_first_record_with_data_in_attribute(attribute='number')
+        self.assertNotEqual(data_to_filter_with, False)
+        self.base_selenium.LOGGER.info('filter with {}'.format(data_to_filter_with))
+        self.test_unit_page.apply_filter_scenario(filter_element='test_unit:testunit_number_filter', filter_text=data_to_filter_with, field_type='text')
+        table_records = self.test_unit_page.result_table()
+        counter = 0
+        while counter < (len(table_records) -1):
+            row_data = self.base_selenium.get_row_cells_dict_related_to_header(row=table_records[counter])
+            self.assertEqual(row_data['Test Unit No.'].replace("'",""), str(data_to_filter_with))
+            counter = counter +1
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test04_testunits.py]: Replace while loop with
```
for table_recodr in table_records
```

File: ui_testing/testcases/basic_tests/test04_testunits.py [hunk 2]
```diff
@@ -1719,4 +1705,168 @@ def test041_hide_all_table_configurations(self):
         """
         assert (self.test_unit_page.deselect_all_configurations(), False)
 
-        
+
+    def test042_filter_by_testunit_no_returns_only_correct_results(self):
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test04_testunits.py]: parametrized 
test042
test045
test046
test047

File: ui_testing/testcases/basic_tests/test04_testunits.py [hunk 3]
```diff
@@ -1719,4 +1705,168 @@ def test041_hide_all_table_configurations(self):
...
+        self.test_unit_page.apply_filter_scenario(filter_element='test_unit:spec_unit_filter', filter_text=data_to_filter_with, field_type='text')
+        table_records = self.test_unit_page.result_table()
+        counter = 0
+        while counter < (len(table_records) -1):
+            row_data = self.base_selenium.get_row_cells_dict_related_to_header(row=table_records[counter])
+            self.assertEqual(row_data['Unit'].replace("'",""), str(data_to_filter_with.replace('{','').replace('}','').replace('[', '').replace(']','')))
+            counter = counter +1
+
+    
+    def test044_filter_by_testunit_category_returns_only_correct_results(self):
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test04_testunits.py]: parameterized
test044
test049
test050

Comment (mazenmmbadr) [reply ui_testing/testcases/basic_tests/test04_testunits.py]: done

File: ui_testing/testcases/basic_tests/test04_testunits.py (hunk 2 above)
Comment (mazenmmbadr) [reply ui_testing/testcases/basic_tests/test04_testunits.py]: done

File: ui_testing/testcases/basic_tests/test04_testunits.py (hunk 1 above)
Comment (mazenmmbadr) [reply ui_testing/testcases/basic_tests/test04_testunits.py]: done

**Pull request merged**